src/fava/beans/funcs.py

```diff
@@ -38,16 +43,10 @@ def get_position(entry: Directive) -> tuple[str, int]:
     raise ValueError(msg)
 
 
-def execute_query(
-    query_: str,
-    entries: list[Directive],
-    options_map: BeancountOptions,
-) -> QueryResult:
+def execute_query(query_: str) -> QueryResult:
     """Execture a query."""
     return query_execute.execute_query(  # type: ignore[no-any-return]
-        query_,
-        entries,
-        options_map,
+        query_
```

Comment:
I most likely don't understand this change: it removes required parameters from the `query_execute.execute_query()` function call. However, as I was writing above, this API is deprecated. The recommended way to execute queries is via the DB-API compatible API:
```python
conn = beanquery.conncet('beancount://path/to/beancount.ledger')
conn.execute(query)
```
In this way, the `beanquery.Connection` object can be reused for multiple queries. If you already have an `entries` list and an `options` map, you can initialize the connection like this:
```python
conn = beanquery.connect('beancount:', entries=entries, options=options)
```

Reply:
> I most likely don't understand this change: it removes required parameters from the query_execute.execute_query() function call. 

Not sure about this either. It's definitely typed incorrectly, it expects a compiled query, which seems to be bound to entries, options and errors already. Will adjust the usage sites to use the connection instead.

Reply:
Sorry, I misread this to be a call to `beaquery.query.run_query()`

Reply:
> it removes required parameters from the query_execute.execute_query() function call.

It actually doesn't, that function only takes a single query parameter now which is why I had to make this change, see https://github.com/beancount/beanquery/blob/master/beanquery/query_execute.py#L121

Reply:
I've removed all usages of execute_query